### PR TITLE
Add Ceph adapter for S3 compatible interfaces

### PR DESCRIPTION
**Pull request by click2cloud-alpha**
This PR adds Ceph adapter for S3 compatible interfaces

### CODE REVIEW

**Comment by leonwanghui** [comment]
@click2cloud-alpha Good job, have you tested the e2e feature? And BTW some unit tests may be also required.

File: api/pkg/s3/datastore/ceph/ceph.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+	"encoding/base64"
+	"fmt"
+
+	"io"
+	"io/ioutil"
+	"time"
+
+	"github.com/Click2Cloud/s3client"
+	. "github.com/Click2Cloud/s3client"
+	"github.com/Click2Cloud/s3client/models"
```
Comment (xing-yang) [inline api/pkg/s3/datastore/ceph/ceph.go]: Thanks for the PR.  Any reason why these can’t be added to OpenSDS repo directly?

Comment (click2cloud-alpha) [reply api/pkg/s3/datastore/ceph/ceph.go]: Do you mean adding this s3client under opensds org?

**Comment by click2cloud-alpha** [comment]
@leonwanghui if you mean testing end to end, then yes we have tested all interfaces except ListParts. Did you mean something else by e2e?
For unit tests, is there any existing location in the repo for unit tests for supported adapters?

**Comment by leonwanghui** [comment]
No, just the same with what you mentioned, and for unit test, you may need to create some test files such like `api/pkg/s3/datastore/ceph/ceph_test.go`

File: api/pkg/s3/datastore/ceph/ceph.go (hunk 1 above)
Comment (xing-yang) [reply api/pkg/s3/datastore/ceph/ceph.go]: Yes

Comment (Savasw) [reply api/pkg/s3/datastore/ceph/ceph.go]: @leonwanghui @xing-yang shall i go ahead with changing the repo to opensds in vendor and ceph.go?

Comment (click2cloud-alpha) [reply api/pkg/s3/datastore/ceph/ceph.go]: @leonwanghui @xing-yang shall i go ahead with changing the repo to opensds in vendor and ceph.go?

**Comment by xing-yang** [comment]
I see what Leon means.  These are all forked from https://github.com/webrtcn/s3client.  Please use that directly.

File: api/pkg/s3/datastore/ceph/ceph.go (hunk 1 above)
Comment (xing-yang) [reply api/pkg/s3/datastore/ceph/ceph.go]: Please use https://github.com/webrtcn/s3client directly and only add additional changes to OpenSDS.

**Comment by sfzeng** [comment]
LGTM except what Xing and wanghui alread mentioned.

**Comment by xing-yang** [comment]
Thanks for addressing the comments about s3client.  Please add unit tests as Leon suggested.

**Comment by xing-yang** [comment]
I'm merging this now.  Please add unit test in a followup PR.  Thanks.

**Pull request merged**